kudu/src/main/java/org/apache/apex/malhar/kudu/partitioner/AbstractKuduInputPartitioner.java

```diff
@@ -0,0 +1,233 @@
+      aMeta.setTotalSize(totalPartitionCount);
+      returnList.add(aMeta);
+      LOG.info("A planned scan meta of the total partitions " + aMeta);
+    }
+    LOG.info("Total kudu partition size is " + returnList.size());
+    return returnList;
+  }
+
+  /***
+   * Clones the original operator and sets the partition pie assignments for this operator. Kryo is used for cloning
+   * @param scanAssignmentsForThisPartition The partition pie that is assigned to the operator according to the
+   *                                        configured partitioner
+   * @return The partition that is used by the runtime to launch the new physical operator instance
+   */
+  public Partitioner.Partition<AbstractKuduInputOperator> clonePartitionAndAssignScanMeta(
+      List<KuduPartitionScanAssignmentMeta> scanAssignmentsForThisPartition)
+  {
+    Partitioner.Partition<AbstractKuduInputOperator> clonedKuduInputOperator =
+        new DefaultPartition<>(KryoCloneUtils.cloneObject(prototypeKuduInputOperator));
+    clonedKuduInputOperator.getPartitionedInstance().setPartitionPieAssignment(scanAssignmentsForThisPartition);
+    return clonedKuduInputOperator;
+  }
+
+  /***
+   * Implements the main logic for defining partitions. The logic to create a partition is pretty straightforward. A
+   * SELECT * expression is used to generate a plan ( a list of kudu scan tokens). The list is then evenly
+   * distributed to each of the physical operator instances as defined by the configuration. At runtime when a new query
+   * needs to be processed, a plan specific to the query is generated and the planned list of kudu scan tokens
+   * for that query are distributed among the available operator instances. Note that we define the partitions only once
+   * during the first invocation. This is because t is ideal we do not optimize on the current query that is running
+   * before we can dynamically partition and cause too much throttling based on the query patterns. Please see
+   * {@link AbstractKuduPartitionScanner#preparePlanForScanners(SQLToKuduPredicatesTranslator)} for details as to how
+   * a query based planning is done.
+   * @param partitions The current set of partitions
+   * @param context The partitioning context as provided by the runtime platform.
+   * @return The collection of planned partitions as implemented by the partitioner that is configured by the user
+   */
+  @Override
+  public Collection<Partition<AbstractKuduInputOperator>> definePartitions(
+      Collection<Partition<AbstractKuduInputOperator>> partitions, PartitioningContext context)
+  {
+    List<Partition<AbstractKuduInputOperator>> partitionsForInputOperator = new ArrayList<>();
+    List<KuduPartitionScanAssignmentMeta> assignmentMetaList = new ArrayList<>();
+    try {
+      assignmentMetaList.addAll(getListOfPartitionAssignments(partitions,context));
+    } catch (Exception e) {
+      LOG.error(" Aborting partition planning as Kudu meta data could not be obtained");
```

Comment:
Why not wrap the exception with this message instead of logging it?

Reply:
Done.